app/models/physical_storage.rb

```diff
@@ -9,12 +9,12 @@ class PhysicalStorage < ApplicationRecord
   has_one :asset_detail, :as => :resource, :dependent => :destroy, :inverse_of => false
 
   has_many :canisters, :dependent => :destroy, :inverse_of => false
-  has_many :computer_system, :through => :canisters
-  has_many :hardware, :through => :computer_system
-  has_many :guest_devices, :through => :canisters
-
   has_many :physical_disks, :dependent => :destroy, :inverse_of => :physical_storage
 
+  has_one :computer_system, :as => :managed_entity, :dependent => :destroy
```

Comment:
You're still missing the previous relations (:hardware, :guest_devices)

IMO this should go back to before #17706 and just add `has_many :canisters` and `has_many :canister_computer_systems`

Reply:
@agrare Fixed it, and put `:guest_devices` through `:hardware` in order to be more general for other providers.